.github/workflows/publish_docker.yml

```diff
@@ -17,13 +17,21 @@ on:
 
 jobs:
   publish_docker_image:
-    name: Publish Docker Image
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - name: Publish image to Dockerhub
+      - name: Install qemu (required for cross-platform builds)
+        run: |
+          sudo apt update
+          sudo apt -y install qemu-system qemu-user-static
+          sudo systemctl restart docker
+      - name: Create multiplatform docker builder
+        run: docker buildx create --use
+      - name: Build and publish images to DockerHub
         env:
           DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
           DOCKER_PASS: ${{ secrets.DOCKER_PASS }}
           DOCKER_REPO: "avaplatform/subnet-evm"
-        run: .github/workflows/publish_docker_image.sh ${{ inputs.vm_id }}
+          PUBLISH: 1
+          PLATFORMS: "linux/amd64,linux/arm64"
+        run: scripts/build_docker_image.sh ${{ inputs.vm_id }}
```

Comment:
It might be interesting to migrate to https://github.com/docker/build-push-action eventually 🤔 
For image tagging, you could also use https://github.com/docker/metadata-action (let me know I can help with it 😉)

Reply:
Hmm this looks good, but looks like it would be a whole new thing. We need to move few parameters like AVALANCHEGO_NODE_IMAGE etc. Feel free to open a new PR.